api/v1alpha1/addonmetadata_types.go

```diff
@@ -127,21 +127,66 @@ type AddonMetadataSpec struct {
 
 	// +optional
 	// OCM representation of an add-on parameter
-	AddOnParameters *ocmv1.AddOnParameterList `json:"addOnParameters"`
+	AddOnParameters *[]ocmv1.AddOnParameter `json:"addOnParameters"`
 
 	// +optional
 	// OCM representation of an addon-requirement
-	AddOnRequirements *ocmv1.AddOnRequirementList `json:"addOnRequirements"`
+	AddOnRequirements *[]ocmv1.AddOnRequirement `json:"addOnRequirements"`
 
 	// +optional
 	// OCM representation of an add-on sub operator. A sub operator is an
 	// operator who's life cycle is controlled by the add-on umbrella operator.
-	SubOperators *ocmv1.AddOnSubOperatorList `json:"subOperators"`
+	SubOperators *[]ocmv1.AddOnSubOperator `json:"subOperators"`
 
 	// +optional
 	// A string which specifies the imageset to use. Can either be 'latest' or a version string
 	// MAJOR.MINOR.PATCH
 	ImageSetVersion *string `json:"addonImageSetVersion"`
+
+	// +optional
+	HasExternalResources *bool `json:"hasExternalResources"`
+
+	// +optional
+	AddonNotifications *[]ocmv1.Notification `json:"addonNotifications"`
+
+	// +optional
+	ManualInstallPlanApproval *bool `json:"manualInstallPlanApproval"`
+
+	// +optional
+	PullSecret string `json:"pullSecret"`
+
+	// +optional
+	// Labels to be applied to all objects created in the SelectorSyncSet.
+	CommonLabels *map[string]string `json:"commonLabels"`
+
+	// +optional
+	// Annotations to be applied to all objects created in the SelectorSyncSet.
+	CommonAnnotations *map[string]string `json:"commonAnnotations"`
+
+	// +optional
+	// Configuration parameters to be injected in the ServiceMonitor used for federation. The target prometheus server found by matchLabels needs to serve service-ca signed TLS traffic (https://docs.openshift.com/container-platform/4.6/security/certificate_types_descriptions/service-ca-certificates.html), and it needs to be runing inside the monitoring.namespace, with the service name 'prometheus'.
+	Monitoring *ocmv1.Monitoring `json:"monitoring"`
```

Comment:
I would create another package for this subtype. It is not currently part of `ocmv1` API per say. We can see what is in OCM API spec (for addons) here: https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/blob/master/pkg/api/addons.go

Maybe create a `mtsrev1` package for all these custom types located at `pkg/mtsre/v1/addons.go` ? WDYT?

Reply:
Great point! will do!